### PR TITLE
Integrate Analytics UI with the analytics backend (EXPOSUREAPP-4823)

### PR DESCRIPTION
**Pull request by kolyaopahle**
### Description
This PR does the last few wiring steps to integrate the analytics backend into the frontend ui

See these two commits for the acutal changes in this PR: https://github.com/corona-warn-app/cwa-app-android/pull/2330/files/aa402778a560b9c7a4cab3b7bdd67c97c3e7ff59..a2f58ac213e08b275d17b73fe67b01a5fe4b690f

### Steps to reproduce
1. Start the app
2. Proceed trough onboarding (enable/disable analytics)
3. Check the settings to see if status is correct
4. Check if analytics submission happens and device follows the abort conditions

### CODE REVIEW

**Review by axelherbstreith — DISMISSED** [review]
LGTM 👍  tested on Samsung Galaxy S8 and Huawei P30

**Review by axelherbstreith — APPROVED** [review]

**Review by chris-cwa — APPROVED** [review]
lgtm!

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/settings/analytics/SettingsPrivacyPreservingAnalyticsState.kt [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    fun getDistrictRowVisibility() =
+        federalState != PpaData.PPAFederalState.FEDERAL_STATE_UNSPECIFIED
+
+    fun getDistrictRowBodyText(context: Context) =
+        district?.districtName ?: context.getString(R.string.analytics_userinput_district_unspecified)
+
+    fun getFederalStateRowBodyText(context: Context) =
+        context.getString(federalState.labelStringRes)
+
+    fun getSettingsPpaSwitchRowState() =
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/settings/analytics/SettingsPrivacyPreservingAnalyticsState.kt:27]: I think it is better here to call it `isSettingsPpaSwitchOn`. it is not actually getting a state. it is just boolean

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/settings/analytics/SettingsPrivacyPreservingAnalyticsState.kt [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+data class SettingsPrivacyPreservingAnalyticsState(
+    val isAnalyticsEnabled: Boolean,
+    val ageGroup: PpaData.PPAAgeGroup,
+    val federalState: PpaData.PPAFederalState,
+    val district: Districts.District?
+) {
+    fun getAgeGroupRowBodyText(context: Context) =
+        context.getString(ageGroup.labelStringRes)
+
+    fun getDistrictRowVisibility() =
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/settings/analytics/SettingsPrivacyPreservingAnalyticsState.kt:18]: `isDistrictRowVisible()`?

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/settings/analytics/SettingsPrivacyPreservingAnalyticsViewModel.kt [hunk 3]
```diff
@@ -1,36 +1,47 @@
...
+import kotlinx.coroutines.flow.first
 import kotlinx.coroutines.flow.flow
 
 class SettingsPrivacyPreservingAnalyticsViewModel @AssistedInject constructor(
+    dispatcherProvider: DispatcherProvider,
     analyticsSettings: AnalyticsSettings,
-    val districts: Districts,
-    dispatcherProvider: DispatcherProvider
+    private val analytics: Analytics,
+    val districts: Districts
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/settings/analytics/SettingsPrivacyPreservingAnalyticsViewModel.kt:20]: `private val districts` ?

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/settings/start/SettingsFragmentViewModel.kt [hunk 4]
```diff
@@ -44,12 +45,10 @@ class SettingsFragmentViewModel @AssistedInject constructor(
...
 
-    // TODO Will be changed by @kolyaophale
-    var analyticsEnabled = false
-
-    var analyticsState: LiveData<SettingsPrivacyPreservingAnalyticsState> = MutableStateFlow(analyticsEnabled)
-        .map { SettingsPrivacyPreservingAnalyticsState((it)) }
-        .asLiveData(dispatcherProvider.Default)
+    var analyticsState: LiveData<SettingsPrivacyPreservingAnalyticsState> =
+        analytics.isAnalyticsEnabledFlow()
+            .map { SettingsPrivacyPreservingAnalyticsState((it)) }
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/settings/start/SettingsFragmentViewModel.kt:50]: is double braces needed in `((it))`?

Comment (kolyaopahle) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/settings/start/SettingsFragmentViewModel.kt]: no, also removed in the line above

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/settings/analytics/SettingsPrivacyPreservingAnalyticsViewModel.kt (hunk 3 above)
Comment (kolyaopahle) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/settings/analytics/SettingsPrivacyPreservingAnalyticsViewModel.kt]: no, but `district` :D

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/settings/analytics/SettingsPrivacyPreservingAnalyticsState.kt (hunk 2 above)
Comment (kolyaopahle) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/settings/analytics/SettingsPrivacyPreservingAnalyticsState.kt]: 👍

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/settings/analytics/SettingsPrivacyPreservingAnalyticsState.kt (hunk 1 above)
Comment (kolyaopahle) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/settings/analytics/SettingsPrivacyPreservingAnalyticsState.kt]: 👍

**Pull request merged**